connector/signaltometricsconnector/config/config.go

```diff
@@ -145,11 +145,15 @@ type MetricInfo struct {
 	// attribute is included in the list then all attributes are included.
 	// Note that configuring this setting might cause the produced metric
 	// to lose its identity or cause identity conflict.
-	IncludeResourceAttributes []Attribute           `mapstructure:"include_resource_attributes"`
-	Attributes                []Attribute           `mapstructure:"attributes"`
-	Histogram                 *Histogram            `mapstructure:"histogram"`
-	ExponentialHistogram      *ExponentialHistogram `mapstructure:"exponential_histogram"`
-	Sum                       *Sum                  `mapstructure:"sum"`
+	IncludeResourceAttributes []Attribute `mapstructure:"include_resource_attributes"`
+	Attributes                []Attribute `mapstructure:"attributes"`
+	// Conditions are a set of OTTL condtions with either AND or OR operator.
+	// Data is processed into metrics only if the sequence evaluates to true.
+	// The condition operator defaults to OR if none specified.
+	ConditionSequence    ConditionSequence     `mapstructure:"condition_sequence"`
```

Comment:
Alternatively: do we need the logic operation right now? If not, we could open an issue upstream, as I think it applies whenever conditions are configurable.

Reply:
I kept going back between the two (I also initially started with the latter). I agree with you, let me fix it.

Reply:
> Alternatively: do we need the logic operation right now? If not, we could open an issue upstream, as I think it applies whenever conditions are configurable.

You could technically use `and` or `or` directly in the statements too like a condition can be `expr_1 and expr_2` so they are not that critical.

Reply:
The countconnector for example doesn't allow configurable logic operation and conditions are ORed by default.

Reply:
I have updated the PR to separate the logic operation. I don't have a strong feeling against removing logic operation too but I kept it for convenience in defining more readable conditions, let me know if you prefer not having that (similar to count connector).

Reply:
I don't have a very strong opinion, but I'd be inclined to remove it for consistency. Another option (which I don't think we should implement now, but raising for discussion) would be to extend `conditions` so that it's not just a list of strings, but could have a tree structure. e.g.

```yaml
conditions:
  and:
    - foo
    - bar
    - or:
      - baz
      - qux
```

Which would translate to `(foo) and (bar) and (baz or qux)`

Reply:
> I don't have a very strong opinion, but I'd be inclined to remove it for consistency.

👍, done!

> Another option (which I don't think we should implement now, but raising for discussion) would be to extend conditions so that it's not just a list of strings, but could have a tree structure. e.g.

I am not sure about this, it feels more complicated to me to grok than the full expression.